### ISSUE TITLE
`--with_tracking`  doesn't seem to work

### ISSUE THREAD

**Issue by Ericmututu** [huggingface/transformers#19082]
### System Info

Hi,
When I enable --with_tracking within run_glue_no_trainer.py, nothing seems to happen. After the training is over, I don't find any log files in output_dir. How do I save these training results (e.g., accuracy, training_loss, ...) as shown in the following figure?
![image](https://user-images.githubusercontent.com/49343975/190840432-b9a2559c-17b3-4653-838d-97c1d3cb56d0.png)
Thanks in advance!

### Who can help?

@sgugger @muellerzr 

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

`Execute the script`

python run_glue_no_trainer.py \
    --model_name_or_path bert-large-uncased \
    --task_name mrpc \
    --output_dir ./output_dir \
    --per_device_train_batch_size 32 \
    --per_device_eval_batch_size 32 \
    --learning_rate 2e-5 \
    --num_train_epochs 100 \
    --seed 42 \
    --with_tracking

### Expected behavior

I want to save the results of the training process in a log file by enabling `--with_tracking`.

**Comment by sgugger** [comment]
@muellerzr will correct me if I'm wrong, but this API is for external trackers (TensorBoard, WandB etc.). To save results on disk, just use regular python code like `json.dump`.

**Comment by muellerzr** [comment]
@sgugger exactly. @Ericmututu do you have any tracking libraries installed on your system?

(I can probably raise an error in the scripts if it's tried and non are available)
